File: scriptingcsharp/vb/Interfaces/Interfaces.vb
Public Interface IScript

    Sub Initialize(ByVal Host As IHost)

    Sub Method1()
    Sub Method2()
    Sub Method3()
    Sub Method4()

End Interface

Public Interface IHost

    ReadOnly Property TextBox() As System.Windows.Forms.TextBox
    Sub ShowMessage(ByVal Message As String)

End Interface

File: scriptingcsharp/csharp/Interfaces/Interfaces.cs
using System;

namespace Interfaces
{
	public interface IScript
	{
		void Initialize(IHost Host);

		void Hail();
		void ParameterChanged(string parameter, string value);
		void Method3();
		void Method4();
	}

	public interface IHost
	{
		System.Windows.Forms.TextBox TextBox { get; }
		void ShowMessage(string Message);
        void SetParameter(string name, string value);
        string GetParameter(string name);

	}
}
